lib/active_fedora/nested_attributes.rb

```diff
@@ -181,8 +183,8 @@ def raise_nested_attributes_record_not_found(association_name, record_id)
       raise ObjectNotFoundError, "Couldn't find #{reflection.klass.name} with ID=#{record_id} for #{self.class.name} with ID=#{id}"
     end
 
-    def call_reject_if(association_name, attributes)
-      return false if has_destroy_flag?(attributes)
+    def call_reject_if(association_name, attributes, allow_destroy)
```

Comment:
This doesn't reject the record. It only skips the reject_if check, as in reject_if returning false so it won't be rejected. The problem here was that reject_if check could be skipped by setting the destroy flag even if the destroy flag wouldn't actually destroy the record.